### PR TITLE
fix #1180: Add missing docs for ConnectOptions

### LINKED ISSUE #1180: docs for `ConnectOptions::retry_on_initial_connect`
### Proposed change

This function should be documented.

### Use case

IMHO this function could use some documentation.

### Contribution

yes

### PR DESCRIPTION
**Pull request by barafael**
Docs for `ConnectOptions::retry_on_initial_connect` and `ConnectOptions::ignore_discovered_servers`.

### CODE REVIEW

File: async-nats/src/options.rs [hunk 1]
```diff
@@ -819,11 +819,20 @@ impl ConnectOptions {
         self
     }
 
+    /// By default, client construction will fail if
```
Comment (Jarema) [inline async-nats/src/options.rs:822]: I would rephrase it `connect` will return error if ..

File: async-nats/src/options.rs [hunk 2]
```diff
@@ -819,11 +819,20 @@ impl ConnectOptions {
...
+    /// the connection to the server cannot be established.
+    ///
+    /// Setting `retry_on_initial_connect` causes client construction
```
Comment (Jarema) [inline async-nats/src/options.rs:825]: It's not exactly how it works. In this case, the handle to the `Client` is returned immediately and conection is handled asynchronously in the background.  Otherwise users would not know the reason for initial connection failure.

File: async-nats/src/options.rs [hunk 3]
```diff
@@ -819,11 +819,20 @@ impl ConnectOptions {
...
+    /// to succeed regardless: in case of initial connection failure,
+    /// the client will periodically attempt connection
+    /// (see [`ConnectOptions::event_callback`]).
     pub fn retry_on_initial_connect(mut self) -> ConnectOptions {
         self.retry_on_initial_connect = true;
         self
     }
 
+    /// By default, a server may advertise other servers in the cluster known to it.
+    /// By setting this option, a connection to these servers will not be attempted.
```
Comment (Jarema) [inline async-nats/src/options.rs:835]: Probably easier to state that client will ignore those advertised servers.
Also adding a short note why it could be useful would be nice - meaning that client might not be able to reach other servers advertised URLs for some reasons.

**Review by Jarema — CHANGES_REQUESTED** [review]
Thank you for your PR!

Some review notes.

File: async-nats/src/options.rs (hunk 2 above)
Comment (barafael) [reply async-nats/src/options.rs]: I'm not sure I understand - if you don't mind, have a look again.

**Review by Jarema — APPROVED** [review]
LGTM!

Thank you for your contribution!
